tests/parser_test.go

```diff
@@ -0,0 +1,78 @@
+package tests
```

Comment:
don't have a separate tests package, but let the tests be with the source code. You can have a different package like "internal_test" if you want to limit to public functions.